test/e2e/network/service.go

```diff
@@ -1682,10 +1682,37 @@ var _ = common.SIGDescribe("Services", func() {
 
 		service := t.BuildServiceSpec()
 		service.Spec.Type = v1.ServiceTypeNodePort
-
+		numberOfRetries := 5
 		ginkgo.By("creating service " + serviceName + " with type NodePort in namespace " + ns)
-		service, err := t.CreateService(service)
-		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
+		var err error
+		for i := 0; i < numberOfRetries; i++ {
+			port, ok := e2eservice.GetUnusedStaticNodePort()
+			if !ok {
+				framework.Logf("Static node port allocator was not able to reserve nodeport so we use the one assigned by the apiserver")
+			} else {
+				service.Spec.Ports[0].NodePort = int32(port)
+			}
```

Comment:
if you made `GetUnusedStaticNodePort` return `(int32, error)` instead of `(int, bool)`, this could be:

```suggestion
			service.Spec.Ports[0].NodePort, err = e2eservice.GetUnusedStaticNodePort()
			framework.ExpectNoError(err, "Could not allocate static NodePort")
```